### ISSUE TITLE
Bug: insecure random number generator mt_rand() used in random_string()

### ISSUE THREAD

**Issue by fliodhais** [codeigniter4/CodeIgniter4#7326]
### PHP Version

7.4

### CodeIgniter4 Version

4.1.1

### CodeIgniter4 Installation Method

Manual (zip or tar.gz)

### Which operating systems have you tested for this bug?

Windows, Linux

### Which server did you use?

apache

### What happened?

I found mt_rand() which is insecure as per [documentation](https://www.php.net/manual/en/function.mt-rand.php) 

> Caution
> This function does not generate cryptographically secure values, and must not be used for cryptographic purposes, or purposes that require returned values to be unguessable.
> 
> If cryptographically secure randomness is required, the [Random\Randomizer](https://www.php.net/manual/en/class.random-randomizer.php) may be used with the [Random\Engine\Secure](https://www.php.net/manual/en/class.random-engine-secure.php) engine. For simple use cases, the [random_int()](https://www.php.net/manual/en/function.random-int.php) and [random_bytes()](https://www.php.net/manual/en/function.random-bytes.php) functions provide a convenient and secure API that is backed by the operating system’s CSPRNG.

Since the function description suggests it as "Useful for generating passwords or hashes.", it should use secure random number generator.

```
	/**
	 * Create a Random String
	 *
	 * Useful for generating passwords or hashes.
	 *
	 * @param string  $type Type of random string.  basic, alpha, alnum, numeric, nozero, md5, sha1, and crypto
	 * @param integer $len  Number of characters
	 *
	 * @return string
	 */
	function random_string(string $type = 'alnum', int $len = 8): string
```

```
			case 'md5':
				return md5(uniqid((string) mt_rand(), true));
			case 'sha1':
				return sha1(uniqid((string) mt_rand(), true));
			case 'crypto':
				return bin2hex(random_bytes($len / 2));
		}
		// 'basic' type treated as default
		return (string) mt_rand();
```

### Steps to Reproduce

system\Helpers\text_helper.php ln 570-585

### Expected Output

```
	case 'md5':
		return md5(uniqid((string) random_bytes(16), true));
	case 'sha1':
		return sha1(uniqid((string) random_bytes(16), true));
	case 'crypto':
		return bin2hex(random_bytes($len / 2));
}
// 'basic' type treated as default
return (string) random_bytes(16);
```

Edit: line number on repo

**Comment by kenjis** [comment]
At least, "Useful for creating passwords or generating random hashes." is wrong.

**Comment by kenjis** [comment]
I sent a PR to fix the documentation. #7333

**Comment by kenjis** [comment]
Updated the user guide:
https://codeigniter4.github.io/CodeIgniter4/helpers/text_helper.html#random_string

**Comment by kenjis** [comment]
> basic: A random number based on mt_rand() (length ignored).
md5: An encrypted random number based on md5() (fixed length of 32).
sha1: An encrypted random number based on sha1() (fixed length of 40).
https://codeigniter4.github.io/CodeIgniter4/helpers/text_helper.html#random_string

What are these types used for? Do they really need to be cryptographically secure?

**Comment by kenjis** [comment]
Ref https://github.com/bcit-ci/CodeIgniter/commit/25ae2a3c0a8f20098181e20d55a61c0c42559ee9

**Comment by kenjis** [comment]
See https://forum.codeigniter.com/showthread.php?tid=87026

**Comment by kenjis** [comment]
Closed by #7333
